documentation/test-quickstart.md

```diff
@@ -0,0 +1,330 @@
+
+```typescript
+import { createTestCredential } from "@azure-tools/test-credential";
+
+const credential = createTestCredential();
+
+// Create your client using the test credential.
+new MyServiceClient(<endpoint>, credential);
+```
+
+To avoid storing the sensitive info in the recordings like authenticating with your Azure endpoints, keys, secrets, etc, we use the sanitizers to mask the values with the fake ones or remove them, `RecorderStartOptions` helps us here. In our generated sample file we have below sanitizers' code:
+
+```typescript
+const envSetupForPlayback: Record<string, string> = {
+  ENDPOINT: "https://endpoint",
+  AZURE_CLIENT_ID: "azure_client_id",
+  AZURE_CLIENT_SECRET: "azure_client_secret",
+  AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
+};
+
+const recorderEnvSetup: RecorderStartOptions = {
+  envSetupForPlaayback,
+};
+
+//...
+await recorder.start(recorderEnvSetup);
+```
+
+#### API Key Authentication
+
+API key authentication would hit the service's endpoint directly so these traffic will be recorded. It doesn't require any customization in tests. However we must secure the sensitive data and not leak into our recordings, so add a sanitizer to replace your API keys. You could read more on how to add sanitizer at [here](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/test-utils/recorder/README.md).
+
+## Example: Basic Azure service interaction and recording
+
+At the code structure [section](#code-structure) we described we'll generate sample file for you, if you are the first time to write test cases you could grow up your own based on them.
+
+This simple test creates a resource and checks that the service handles it correctly in the project `purview-catalog-rest`. Below are the steps:
+
+- Step 1: Create your test file and add one test case with resource creation, here we have purview catalog glossary test file `glossary.spec.ts` and one case named `Should create a glossary`. Or rename the `sampleTest.spec.ts` file and its case `sample test`.
+- Step 2: Add the utility method `createClient` in `public/utils/recordedClient.ts` to share the `PurviewCatalogClient` creation.
+  - Call `createTestCredential` to init your credential and refer [here](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/test-utils/recorder/MIGRATION.md#aad-and-the-new-noopcredential) for more details
+  - Wrap the `option` with test options by calling `recorder.configureClientOptions(options)`
+- Step 3: In `glossary.spec.ts` file call `createClient` to prepare the client and call `client.path("/atlas/v2/glossary").post()` to create our glossary resource under our case `Should create a glossary`
+- Step 4[Optional]: Specify environment variables that would be faked in the recordings in map `envSetupForPlayback` under the file `public/utils/recordedClient.ts`.
+- Step 5: In `glossary.spec.ts` file add necessary assertions in your test case
+- Step 6: Run and record your test cases
+
+### `glossary.spec.ts`
```

Comment:
should we add some example for mgmt plane test ? as the code for mgmt plane would look differently